romancal/photom/photom.py

```diff
@@ -0,0 +1,169 @@
+        input roman datamodel
+
+    photom_metadata : list
+        Set of matched photom meta data keys
+
+    Returns
+    -------
+
+    """
+    # Get the scalar conversion factor.
+    conversion = photom_metadata['photmjsr']  # unit is MJy / sr
+
+    # Store the conversion factor in the meta data
+    log.info(f'photmjsr value: {conversion:.6g}')
+    input_model.meta.photometry.conversion_megajanskys = conversion
+    input_model.meta.photometry.conversion_microjanskys = conversion.value * MJSR_TO_UJA2 *\
+                                                          u.microjansky / (u.arcsecond**2)
+
+    # Get the scalar conversion uncertainty factor
+    uncertainty_conv = photom_metadata['uncertainty']
+
+    # Store the uncertainty conversion factor in the meta data
+    log.info(f'uncertainty value: {conversion:.6g}')
+    input_model.meta.photometry.conversion_megajanskys_uncertainty = uncertainty_conv
+    input_model.meta.photometry.conversion_microjanskys_uncertainty = uncertainty_conv.value * \
+                                                  MJSR_TO_UJA2 * u.microjansky / (u.arcsecond**2)
+
+    # Return updated input model
+    return input_model
+
+
+def find_photom_parameters(photom_table, match_fields):
+    """
+    Find metadata keyword object matching fields.
+
+    Parameters
+    ----------
+    photom_table : dict
+        "Optical Element":
+            {"photmjsr": value * u.MJ / u.sr,
+             "uncertainty": value* u.MJ / u.sr,
+             "pixelareasr": value * u.sr}
+    match_fields : dict
+        {field_name: value} pair to use as a matching criteria.
+
+    Raises
+    ------
+    Warning
+        When a field name is not in the table.
+    MatchFitsTableRowError
+        When more than one rows match.
+
+    Returns
+    -------
+    row : dict
+        Matched metadata keyword object
+    """
+
+    # Find photom entries that match the optical element in match_fields
+    results = [
+        item[1].upper() in dict(photom_table).keys() for item in
+        match_fields.items()
+    ]
+
+    # Trim to matched rows and test that we have exactly one match
+    photom_parameters_match = list(filter(bool, results))
+    if len(photom_parameters_match) > 1:
+        warnings.warn(f"Expected to find one matching row in table, found {len(photom_parameters_match)}.")
+    if len(photom_parameters_match) == 0:
+        warnings.warn("Expected to find one matching row in table, found 0.")
+        return None
+
+    # Return photom parameters for the selected optical element
+    return photom_table[match_fields['optical_element'].upper()]
+
+
+
+def save_area_info(input_model, photom_parameters):
+    """
+    Short Summary
+    -------------
+    Read the pixel area value in the photom parameters, then convert and
+    copy them to the metadata of the input datamodel.
+
+    Parameters
+    ----------
+    input_model : Roman level 2 image datamodel
+        input roman datamodel
```

Comment:
nitpick : Capitalize the definition - `Input Roman datamodel` to be consistent

Reply:
Done, and done elsewhere as well.